### DISCUSSION TITLE
Issues with Safari IOS, possibly fixed by pdfjs 3+?

### DISCUSSION THREAD

**Discussion by KtodaZ** [wojtekmaj/react-pdf#1323] [General]
Hello,

My team is attempting to release an IOS mobile app in which we have a "stacked" document view consisting of many pdfs with ability to scroll from one to another, and some editing experience we've built on top. 

Our app works great on desktop, and has for some time, but has strange crashing / page reloading issues when scrolling quickly between many documents or zooming in after scrolling, specifically within IOS Safari (15+). The issue does seem to get worse with more documents, so a potential solution on our end is to limit the number of documents rendered, but we fear large documents may still have issues and would prefer to keep the current UX.

These issues to me point to safari memory management issues, but this is very difficult to debug as there are no crash logs and the safari developer console does not show any errors.

We have virtualization built in, and are at most rendering 3-4 pages at a time, so we don't think there's anything we're doing wrong besides possibly loading so many documents 🙃

In the [latest release](https://github.com/mozilla/pdf.js/releases) of pdfjs it seems that many features and optimizations. Is there an experimental branch we could try out the new version to see if it fixes our issues? 
What's your perceived level of difficulty of upgrading if not?

Also, are there any reports you've seen similar to our issue that you might suggest?

Thank you!

**Comment by kimvale-singlifeph** [comment]
> Discussion options

We have a common issue and I haven't see any thread that offers solutions even from the author.

**Comment by KtodaZ** [comment]
I seem to have found a solution to this issue, and it mostly makes things better, also rendering performance seems to have improved.

Note that this solution is untested as of now but seems okay so far. It doesn't seem like there was an upgrade guide attached to the release of [pdfjs 3.0](https://github.com/mozilla/pdf.js/releases/tag/v3.0.279), so perhaps this change is safe.

I modified package.json with the following, which forces react-pdf to use the latest version of pdfjs:
```
  "overrides": {
    "react-error-overlay": "6.0.9",
    "react-pdf": {
      "pdfjs-dist": "^3.2.146"
    },
    "@types/react-pdf": {
      "pdfjs-dist": "^3.2.146"
    }
  }
```

**Comment by wojtekmaj** [comment]
This is going to be addressed with #1325 but as a breaking change, will be released with a future breaking version.

**Reply by KtodaZ** [reply]
2 days turnaround... amazing work thank you!

**Reply by wojtekmaj** [reply]
Don't get too excited 😆 I'm starting my month long holidays today.

**Reply by KtodaZ** [reply]
Noo! All good enjoy!

**Reply by wojtekmaj** [reply]
Heads up: v7 beta is out 👀

**Reply by ceisele-r** [reply]
@wojtekmaj exciting! Are docs for v7 beta already somewhere around? I could no longer find the `entry.vite` in v7 and also no docs how to use the new hooks.

**Reply by wojtekmaj** [reply]
Docs are both included in the release and available on main branch on GitHub (at the moment of writing, it's constantly under development).

entry.vite is gone - see README.

Not sure what hooks are you referring to.

**Reply by ceisele-r** [reply]
Thank you very much for your fast response, I unfortunately totally overlooked the fact that the Readme already refers to v7. I searched for v7 in this list: https://github.com/wojtekmaj/react-pdf#before-you-continue
My fault, sorry.

Regarding the `entry.vite` replacement, I solved it using this recommendation (I mention it since it might help others): https://github.com/wojtekmaj/react-pdf#import-worker-recommended Thank you!

Regarding the hooks: I interpreted the sentence "Rewritten package using React Hooks" in the changelog https://github.com/wojtekmaj/react-pdf/releases/tag/v7.0.0-beta in a way that `react-pdf` now exports react hooks to be used to display the PDF. But from your answer I guess internally it is now using hooks, which is fine.

Thanks again!

**Reply by Anhaachan** [reply]
Does this mean the error is fixed starting from version 7?
